nautobot/core/views/mixins.py

```diff
@@ -280,6 +281,16 @@ def check_permissions(self, request):
                     code=getattr(permission, "code", None),
                 )
 
+        # Check for drf-specific permissions (IsAutheticated, etc) in permission_classes which is empty by default.
+        # self.get_permissions() iterates through permissions specified in the `permission_classes` attribute.
+        for permission in self.get_permissions():
+            # If the user does not have the permission required, we raise DRF's `NotAuthenticated` or `PermissionDenied` exception
+            # which will be handled by self.handle_no_permission() in the UI appropriately in the dispatch() method
+            if not permission.has_permission(request, self):
+                self.permission_denied(
+                    request, message=getattr(permission, "message", None), code=getattr(permission, "code", None)
+                )
```

Comment:
Would be good to add a unit test or two (possibly using a custom view in the example app?) to make sure this functionality is working as intended.